iroh-net/src/relay/http/client.rs

```diff
@@ -784,10 +828,12 @@ impl Actor {
 
     fn use_https(&self) -> bool {
         // only disable https if we are explicitly dialing a http url
-        if self.url.scheme() == "http" {
-            return false;
+        #[allow(clippy::match_like_matches_macro)]
```

Comment:
why the preference over the suggested form?

Reply:
Clippy suggests this:
```rust
!matches!(self.url.scheme(), "http" | "ws")
```

I personally find it harder to figure out which value this is going to take in which case. I also skip over the first negation really easily. Would you disagree?

Reply:
Orthogonal: Actually reading over this - this reminds me I should rename this function to `use_tls`